packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -29,6 +34,48 @@
   private final FlutterWebViewClient flutterWebViewClient;
   private final Handler platformThreadHandler;
 
+  // Verifies that a url opened by `Window.open` has a secure url.
+  private class FlutterWebChromeClient extends WebChromeClient {
+    @Override
+    public boolean onCreateWindow(
+        final WebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
+      final WebViewClient webViewClient =
+          new WebViewClient() {
+            @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+            @Override
+            public boolean shouldOverrideUrlLoading(
+                @NonNull WebView view, @NonNull WebResourceRequest request) {
+              final String url = request.getUrl().toString();
+              if (isSecure(url)) {
+                flutterWebViewClient.shouldOverrideUrlLoading(FlutterWebView.this.webView, request);
+              }
+              return true;
+            }
+
+            @Override
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+              if (isSecure(url)) {
+                flutterWebViewClient.shouldOverrideUrlLoading(FlutterWebView.this.webView, url);
+              }
+              return true;
+            }
+          };
+
+      final WebView newWebView = new WebView(view.getContext());
```

Comment:
Should we display this new webview?
Should it be a FlutterWebView instance?
Should we destroy the previous webview instance?
Should we re-bind the method channel to work with this new webview?

Reply:
Thanks for explaining this offline!
So the `newWebView` is not displayed, it is instead used to intercept the url that was supposed to be loaded and make the "regular" webview load that url instead.

I'm still a little confused as I expected this logic to trigger `loadUrl` on the main webview which I don't see directly. Moving this discussion to the specific line above.

Reply:
This was based on a suggestion. I will share it with you.  This temporary webview serves as a proxy to get the attempted URL and set it back in the original webview provided it's a safe URL.